src/rest_offsets.lua

```diff
@@ -88,34 +88,33 @@ function make_the_change()
     end
     local current_staff = nil
     local staff_spec = finale.FCCurrentStaffSpec()
-    local rest_type = { "OtherRestPosition", "HalfRestPosition", "WholeRestPosition", "DoubleWholeRestPosition" }
 
     for entry in eachentrysaved(finenv.Region(), config.layer) do
         if entry:IsRest() then
-            if config.offset == 0 then
-                entry:SetFloatingRest(true)
-            else
+            --if config.offset == 0 then
+            --    entry:SetFloatingRest(true)
+            --else
                 if current_staff ~= entry.staff then -- need a new staff spec
                     current_staff = entry.staff
                     staff_spec:LoadForEntry(entry)
                 end
+                local rest_prop = "OtherRestPosition"
                 local duration = entry.Duration
-                if duration % 3 == 0 then -- it's dotted
-                    duration = duration * 2 / 3 -- get the un-dotted version
+                if duration >= finale.BREVE then
+                    rest_prop = "DoubleWholeRestPosition"
+                elseif duration >= finale.WHOLE_NOTE then
+                    rest_prop = "WholeRestPosition"
+                elseif duration >= finale.HALF_NOTE then
+                    rest_prop = "HalfRestPosition"
                 end
-                local power_count = 1
-                while duration > 1024 and power_count <= #rest_type do
-                    duration = duration / 2
-                    power_count = power_count + 1
-                end
-                local rest_type_offset = staff_spec[rest_type[power_count]] + 4 -- adjusted for middle line
+                local total_offset = staff_spec[rest_prop] + config.offset
```

Comment:
This computation is now correct. Basically, the `...RestPosition` properties are *telling* you the middle line. So by adding it in your are offsetting from the middle line. So your text above should remain {-4, 0, 4}. I just tested your code, so I know I'm right. I'm concerned that you haven't seen this in your own testing.

Also, another issue I noticed is that you must reload `FCCurrentStaffSpec` for every entry. Don't try to be smart about it. This is because staff styles can change the shape of the staff in the middle of a measure. That's the whole reason `FCCurrentStaffSpec` exists instead of plain old `FCStaff`. It is very efficient to load, so just do it.